src/Microsoft.Azure.CosmosRepository/Repositories/DefaultRepository.cs

```diff
@@ -62,4 +67,89 @@ await iterator.ReadNextAsync(cancellationToken)
 
         return (results, charge, continuationToken);
     }
+
+    /// <summary>
+    /// Builds a partition key from the first item in a list. This method assumes all items in the list belong to the same partition.
+    /// Throws an exception if the list is empty.
+    /// </summary>
+    /// <param name="items">A list of items from which the partition key is to be derived.</param>
+    /// <returns>A PartitionKey object constructed from the first item in the list.</returns>
+    /// <exception cref="ArgumentException">Thrown when the items list is empty.</exception>
+    internal static PartitionKey BuildPartitionKey(List<TItem> items)
+    {
+        if (items.Count == 0)
+        {
+            throw new ArgumentException(
+                "Unable to perform batch operation with no items",
+                nameof(items));
+        }
+        return BuildPartitionKey(items[0]);
+    }
+
+    /// <summary>
+    /// Constructs a partition key from a collection of string values. If the collection is empty or null, 
+    /// the method uses the defaultValue, if provided, to construct the PartitionKey.
+    /// </summary>
+    /// <param name="values">An IEnumerable collection of string values for constructing the partition key.</param>
+    /// <param name="defaultValue">An optional default value used to construct the PartitionKey if values are null or empty.</param>
+    /// <exception cref="ArgumentException">Thrown when the number of provided partition key values exceeds 3.</exception>
+    /// <returns>A PartitionKey object constructed from the values or the defaultValue if values are empty or null.</returns>
+    internal static PartitionKey BuildPartitionKey(IEnumerable<string> values, string? defaultValue = null)
+    {
+        var builder = new PartitionKeyBuilder();
+        if (values == null || !values.Any())
+        {
+            return !string.IsNullOrEmpty(defaultValue) ? new PartitionKey(defaultValue) : default;
+        }
+
+        if (values.Count() > 3) throw new ArgumentException("Unable to build partition key. The max allowed partition key values is 3", nameof(values));
```

Comment:
```suggestion
        var keys = values?.ToList();
        if (keys is null or keys is { Count: 0 })
        {
            return !string.IsNullOrWhiteSpace(defaultValue)
                ? new PartitionKey(defaultValue)
                : default;
        }

        if (keys.Count > 3)
        {
            throw new ArgumentException(
                "Unable to build partition key. The max allowed number of partition key values is 3.", 
                nameof(values));
        }
```

Reply:
Getting following IDE error, any clues?

```
Severity	Code	Description	Project	File	Line	Suppression State
Error	CS9135	A constant value of type 'List<string>' is expected	Microsoft.Azure.CosmosRepository (net7.0), Microsoft.Azure.CosmosRepository (net8.0), Microsoft.Azure.CosmosRepository (netstandard2.0)	C:\Prohousia\External\azure-cosmos-dotnet-repository\src\Microsoft.Azure.CosmosRepository\Repositories\DefaultRepository.cs	96	Active
```

Reply:
Yeah, I think the pattern matching is a bit off, suggested it through GitHub review UX, no compiler checks here 🤣. Try this:

```cs
if (keys is null or { Count: 0 })
```